### ISSUE TITLE
Signal scale updates on initialization

### ISSUE THREAD

**Issue by arvind** [vega/vega#1433]
I would like to have a signal that computes scaled coordinates **once**, when the scale is initialized with a valid domain and range, and then no longer reacts to any further updates to the scale.

In the following spec, if `react` is `true`, only a single warning is printed which suggests that the signal is correctly marked as dependent on the scale. However, setting `react` to `false` produces a single `[undefined, undefined]` which suggests that the signal is no longer being topologically evaluated after the scales are initialized? It's unclear to me if this is a bug or by design and, if the latter, how I would accomplish my desired outcome. 

```json
{
  "$schema": "https://vega.github.io/schema/vega/v4.json",
  "width": 200,
  "height": 200,
  "data": [
    {
      "name": "source_0",
      "url": "data/cars.json"
    }
  ],
  "signals": [
    {
      "name": "box_x",
      "update": "warn('box_x', [scale(\"x\", 100),scale(\"x\", 150)])",
      "react": false
    }
  ],
  "scales": [
    {
      "name": "x",
      "type": "linear",
      "domain": {"data": "source_0", "field": "Horsepower"},
      "range": [0, {"signal": "width"}]
    }
  ]
}
```

**Comment by jheer** [comment]
I have solutions implemented for both this issue and #1435. However, we should discuss and finalize some corresponding design decisions prior to commit and release.

My current design is to (1) keep the current signal update/react semantics unchanged for backwards compatibility, and (2) introduce a new initialization-only option for signals, exposed in the Vega JSON spec as a new signal `init` property. The `init` property would take a Vega expression string, and the schema would enforce that `init` is mutually exclusive with either `react` or `update`. This design opens up the possibility of deprecating the `react` parameter and only supporting vanilla `init` and `update` modes in the future.

An alternative design would be to keep the Vega JSON spec definition unchanged, but change the semantics of `react: false` to wire up dependencies but then sever them after the initial evaluation.

Both designs actually have the same implementation at the dataflow level, but obviously differ at the JSON spec level. We will need to bump the minor version in any case due to dataflow API extensions. 

@arvind, any opinions here?

**Comment by jheer** [comment]
There are internal use cases for bridging event streams with the dataflow graph, but no JSON-spec level use cases I know of other than initialization.

**Comment by arvind** [comment]
@jheer: A gentle ping on this issue and #1435.

**Comment by jheer** [comment]
Addressed with the `init` signal property, released in Vega v4.4.0.
